client/src/lib/utils/Application/FetchApplication.ts

```diff
@@ -0,0 +1,27 @@
+import {SupabaseClient} from '@supabase/supabase-js';
+import {Profile} from '@/lib/store/profile';
+import {Application} from '@/lib/store/track';
+
+export async function FetchApplication(
+    supabase: SupabaseClient,
+    profile: Profile,
+): Promise<Application[] | undefined> {
+  if (!profile) {
+    console.error('Profile is required to fetch applications');
```

Comment:
I would check if profile exists within the file where you're actually calling fetchApplications. I wouldn't do it within here

Reply:
fixed!